src/app/dataset-view/additional-components/lineage-component/lineage.component.ts

```diff
@@ -108,11 +107,36 @@ export class LineageComponent extends BaseComponent implements OnInit {
                     isCurrent: dataset.id === currentDataset.id,
                     access: "private",
                     accountName: dataset.owner.name,
+                    sourceUrl:
+                        dataset.metadata.currentSource?.fetch.__typename === "FetchStepUrl"
+                            ? dataset.metadata.currentSource.fetch.url
+                            : "Other source",
                 },
             });
         }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        const extraNodes = this.lineageGraphNodes.filter((item: Node) => item.data.kind === DatasetKind.Root);
+        extraNodes.forEach((node) => {
+            this.lineageGraphNodes.push({
+                id: "extra-node-" + node.id,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
```

Comment:
I don't like these pragmas. I understand that library model offers `any` type for the `data` part, but we could desribe our own model interface, and cast when accessing `node.data`. The need in pragmas would than disappear.

Reply:
What will you do if we have a 3rd type of the node next months - rewrite everything?
Differentiating based on `extraNode` boolean is a bad idea, it won't survive more node types.

Perhaps an enum-based variant structure for node data:

```
interface LineageGraphNodeData {
    nodeKind: LineageGraphNodeType;
    nodeDataObject: LineageGraphDatasetNodeObject | LineageGraphSourceNodeObject;
}

enum LineageGraphNodeType {
   Dataset = "dataset",
   Source = "source",
}
```

Then next time, you would just add an enum member + a new node object structure.

Reply:
Fixed